Common/ConfigParser.cs

```diff
@@ -127,6 +133,27 @@ public class UploaderSubsection
             public IncludeExcludePatterns AssemblyPatterns { get; set; }
         }
 
+        /// <summary>
+        /// Contains the thresholds for purging trace files from the upload archives.
+        /// </summary>
+        public class PurgeUploadArchivesSection
+        {
+            /// <summary>
+            /// Minimal timespan to keep uploaded traces.
+            /// </summary>
+            public TimeSpan UploadedTraces { get; set; } = TimeSpan.MinValue;

```

Comment:
For the UploadInterval you used an int for parsing, here you use TimeSpans. Is this on purpose? I'd expect us to use the same type for both.

Does the YAML parser guarantee that ints will always be deserialized into timespans with unit "Days" or is this more or less by accident? In fact, looking over the tests that we have, I don't think we actually verify that. I'd expect that this means that this is a bug because the Timespan will be initialized with the given number but that will be interpreted as "ticks" not as "days". So I'd prefer that we use ints for deserializing and convert to timespans ourselves when we construct the Config from the YamlConfig. That way we should be on the safe side :-)

Reply:
This is not by accident. Json.NET explicitly support [(de)serializing TimeSpans](https://www.newtonsoft.com/json/help/html/SerializeDateTimeZoneHandling.htm). The full format for a TimeSpan is "1.00:00:00.000" (d.hh:mm:ss.mmm). For days the abbreviation to an int is accepted by the parser. For minutes we would could write "00:05:00" in the config, but I didn't like that.

Correct conversion is tested: two tests in `ConfigTest` expect days timespans according to the ints in the config file.